tests/operator/operator.go

```diff
@@ -610,17 +610,14 @@ var _ = Describe("[sig-operator]Operator", Serial, decorators.SigOperator, func(
 			waitForKvWithTimeout(kv, 120)
 
 			By("Check that worker nodes resumed the default amount of allocatable kvm devices")
-			defaultKvmDevices := "1k"
-			Eventually(func() error {
+			const defaultKvmDevices = "1k"
+			defaultKvmDevicesQuant := resource.MustParse(defaultKvmDevices)
+			Eventually(func(g Gomega) {
 				nodesWithKvm := libnode.GetNodesWithKVM()
 				for _, node := range nodesWithKvm {
-					kvmDevices, _ := node.Status.Allocatable[services.KvmDevice]
-					if kvmDevices != resource.MustParse(defaultKvmDevices) {
-						return fmt.Errorf("node %s does not have the expected allocatable kvm devices: %s, got: %d", node.Name, defaultKvmDevices, kvmDevices.Value())
-					}
+					g.Expect(node.Status.Allocatable).To(HaveKeyWithValue(services.KvmDevice, defaultKvmDevicesQuant), "node %s does not have the expected allocatable kvm devices", node.Name)
```

Comment:
I think something like this can be cleaner (return node slice, and assert on each node):
https://github.com/openshift/cluster-api-actuator-pkg/blob/86066d47a2646f407509a9a239124794ce0cefb3/pkg/autoscaler/autoscaler.go#L641-L660

Reply:
I agree in general, but I think in this case, if the assertion will fail, we'll get unreadable error message from gomega, including a partial printout of the whole node list, but it will probably too long to be in the error message, so gomega will cut it, making the error message usefulness for us, and it won't help us to debug.